ruby/main.rb

```diff
@@ -0,0 +1,54 @@
+require "faraday"
+require "json"
+
+API_KEY = "YOUR_API_KEY_HERE" # it should look like this: VF.XXXXXXX.XXXXXX... keep this a secret!
+VERSION_ID = "YOUR_VERSION_ID_HERE" # your voiceflow project versionID, find it under the "integrations" tab
+
+def interact(user_id, request)
+    puts "..."
+    puts user_id, request.to_json
+
+    # call the voiceflow api with the user's name & request, get back a response
+    response = Faraday.post(
+        "https://general-runtime.voiceflow.com/state/#{VERSION_ID}/user/#{user_id}/interact",
+        { 
+            headers: {"Content-Type" => "application/json", "Authorization": API_KEY },
+            body: request 
+        }
+    )
+
+    if response.nil?
+        return false
+    end
+
+    # loop through the response
+    response.data.each do |trace|
+        # if trace.type == "speak"
+        #     puts trace.payload.message
+        # elsif trace.type == 'text'
+        #     puts trace.payload.text
+        # elsif trace.type == 'end'
+        #     # an end trace means the the voiceflow dialog has ended
+        #     return false
+    end
+
+    return true
+end
+
+def main()
```

Comment:
you technically don't even need this `main` function. I only did it in NodeJS because you can't use await in the top level code.